src/core/build/findEntrypoints.ts

```diff
@@ -38,53 +38,46 @@ export async function findEntrypoints(
   relativePaths.sort();
 
   const pathGlobs = Object.keys(PATH_GLOB_TO_TYPE_MAP);
-  const existingNames: Record<string, Entrypoint | undefined> = {};
 
-  const entrypoints: Entrypoint[] = [];
   let hasBackground = false;
-  await Promise.all(
+  // TODO: This parallelization is bad
+  const possibleEntrypoints: Array<Entrypoint | undefined> = await Promise.all(
     relativePaths.map(async (relativePath) => {
```

Comment:
This `Promise.all` was causing race conditions when pushing results to the `entrypoints` array. Broken the data loading out into it's own parallel `Promise.all` and filtered and analyzed separately.